### PR TITLE
Admin login issue fix.

### LINKED ISSUE #32100: [Issue] Admin login issue fix.
This issue is automatically created based on existing pull request: magento/magento2#32099: Admin login issue fix.

---------

### Fixed Issues (if relevant)

1. None

### Manual testing scenarios (*)

1. Install Magento 2 with `--use-rewrites=0` option or set `web/seo/use_rewrites` to `0` in core_config_data_table.
2. Open URL `<magento host>/<backend front name>/` in incognito browser mode.
3. Try to log in as admin.

### Actual result
Not logged in - page `<magento host>/<backend front name>/` reloaded.

### Expected result
Logged in.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by engcom-Foxtrot**
### Related Pull Requests
https://github.com/magento/security-package/pull/291

### Fixed Issues (if relevant)

1. None

### Manual testing scenarios (*)

1. Install Magento 2 with `--use-rewrites=0` option or set `web/seo/use_rewrites` to `0` in core_config_data_table.
2. Open URL `<magento host>/<backend front name>/` in incognito browser mode.
3. Try to log in as admin.

### Actual behavior
Not logged in - page `<magento host>/<backend front name>/` reloaded.

### Expected behavior
Logged in.

### Questions or comments

On the actual 2.4-develop admin session cookie path is generated in \Magento\Backend\Model\Session\AdminConfig::extractAdminPath. This method considers the path part of the base URL. When URL rewrites are disabled, Magento saves admin cookie with `/index.php/<backend front name>` path (see an attached screenshot), so accessing admin login screen by `<magento host>/<backend front name>/` URL does not fit the cookie path.

Screenshot from 2021-02-10 18-05-31

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#32100: Admin login issue fix.

### CODE REVIEW

**Comment by gabrieldagama** [comment]
@magento create issue

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Review by gabrieldagama — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-8784](https://jira.corp.magento.com/browse/ENGCOM-8784) has been created to process this Pull Request
:eight_spoked_asterisk: @gabrieldagama, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by engcom-Bravo — CHANGES_REQUESTED** [review]
:x: QA Failed

Each time after Sign In to Admin Panel, an _Invalid security or form key. Please refresh the page._ error message appears
refresh_errore

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

File: app/code/Magento/Backend/Controller/Adminhtml/Auth/Login.php [hunk 1]
```diff
@@ -69,4 +97,26 @@ private function getRedirect($path)
         $resultRedirect->setPath($path);
         return $resultRedirect;
     }
+
+    /**
+     * Verify if correct backend uri requested.
+     *
+     * @return bool
+     */
+    private function isValidBackendUri(): bool
```
Comment (ihor-sviziev) [inline app/code/Magento/Backend/Controller/Adminhtml/Auth/Login.php:115]: @engcom-Foxtrot 
TBH it's better to cover this functionality with some unit or integration test, functional test is good, but there for sure not a single test case. What do you think?

File: app/code/Magento/Backend/Controller/Adminhtml/Auth/Login.php [hunk 2]
```diff
@@ -69,4 +97,26 @@ private function getRedirect($path)
...
+    {
+        $requestUri = $this->getRequest()->getRequestUri();
+        $backendApp = $this->backendAppList->getCurrentApp();
+        $baseUrl = parse_url($this->backendUrlFactory->create()->getBaseUrl(), PHP_URL_PATH);
+        if (!$backendApp) {
+            $backendFrontName = $this->frontNameResolver->getFrontName();
+        } else {
+            //In case of application authenticating through the admin login, the script name should be removed
+            //from the path, because application has own script.
+            $baseUrl = Http::getUrlNoScript($baseUrl);
```
Comment (ihor-sviziev) [inline app/code/Magento/Backend/Controller/Adminhtml/Auth/Login.php]: can we avoid using static method call here?

Comment (engcom-Foxtrot) [reply app/code/Magento/Backend/Controller/Adminhtml/Auth/Login.php]: @ihor-sviziev I think only by copying this method code.

Comment (ihor-sviziev) [reply app/code/Magento/Backend/Controller/Adminhtml/Auth/Login.php]: @engcom-Foxtrot  Could you explain why can't we create an object via factory and run the method from it?

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Integration Tests, WebAPI Tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Integration Tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8784](https://jira.corp.magento.com/browse/ENGCOM-8784) has been created to process this Pull Request

**Comment by engcom-Oscar** [comment]
:heavy_check_mark: QA passed

Was checked case from Manual testing scenarios 

1. Install Magento 2 with `--use-rewrites=0` option or set `web/seo/use_rewrites` to `0` in core_config_data_table.
2. Open URL `<magento host>/<backend front name>/` in incognito browser mode.
3. Try to log in as admin.

:x: **Before:**
Not logged in - page `<magento host>/<backend front name>/` reloaded.

:heavy_check_mark: **After:**
Logged in.

**Pull request merged**